[release-1.5] Add note on automatic mutual TLS in the upgrade note (#6763)
istio/istio.io: content/en/news/releases/1.5.x/announcing-1.5/upgrade-notes/index.md

* We have stabilized the SDS certificate and key provisioning flow. Now the Istio workloads are using SDS to provision certificates. The secret volume mount approach is deprecated.
* Please note when mutual TLS is enabled, Prometheus deployment needs to be manually modified to monitor the workloads. The details are described in this [issue](https://github.com/istio/istio/issues/21843). This is not required in 1.5.1.

## Automatic mutual TLS
 

> Automatic mutual TLS is now enabled by default. Traffic between sidecars are automatically configured as mutual TLS. You can disable this explicitly if you worry about the encryption overhead by adding the option `-- set values.global.mtls.auto=false` during install. For more details, check the [automatic mutual TLS task](/docs/tasks/security/authentication/auto-mtls/)

Reviewer: miss a "."

> Automatic mutual TLS is now enabled by default. Traffic between sidecars are automatically configured as mutual TLS. You can disable this explicitly if you worry about the encryption overhead by adding the option `-- set values.global.mtls.auto=false` during install. For more details, check the [automatic mutual TLS task](/docs/tasks/security/authentication/auto-mtls/)

Reviewer: question for native speakers, traffic should be uncountable noun, thus "s/are/is"?

Reviewer 2: You are correct it should be `is`.

> Automatic mutual TLS is now enabled by default. Traffic between sidecars are automatically configured as mutual TLS. You can disable this explicitly if you worry about the encryption overhead by adding the option `-- set values.global.mtls.auto=false` during install. For more details, check the [automatic mutual TLS task](/docs/tasks/security/authentication/auto-mtls/)

Reviewer 3: ```suggestion
Automatic mutual TLS is now enabled by default. Traffic between sidecars is automatically configured as mutual TLS. You can disable this explicitly if you worry about the encryption overhead by adding the option `-- set values.global.mtls.auto=false` during install. For more details, refer to [automatic mutual TLS](/docs/tasks/security/authentication/authn-policy/#auto-mutual-tls).
```
